### PR TITLE
Introduce hyperfed command as a wrapper

### PR DESCRIPTION
**Pull request by shashidharatd**
similar to `hyperkube` command in k8s, `hyperfed` command is introduced as a wrapper for controller-manager and kubefed2 binaries. This is according to the comment in https://github.com/kubernetes-sigs/federation-v2/pull/416#issuecomment-438718379

### CODE REVIEW

**Comment by shashidharatd** [comment]
PR https://github.com/kubernetes-sigs/federation-v2/pull/416 itslef is updated with the commits in this pr, hence closing this.

**Comment by shashidharatd** [comment]
oops, i just saw the comment from paul in the other pr. I too feel the commits could be reviewed together as the original code size in that pr was very small and hence closing this for now.

**Comment by marun** [comment]
I'm reviewing here!

File: cmd/hyperfed/main.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+	commandFns := []func() *cobra.Command{
+		controller,
+		kubefed2Cmd,
+	}
+
+	makeSymlinksFlag := false
+	cmd := &cobra.Command{
+		Use:   "hyperfed",
+		Short: "Request a new project",
```
[the remark this answers is not in the document]
Comment (shashidharatd) [reply cmd/hyperfed/main.go]: Updated.

File: cmd/controller-manager/app/controller-manager.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+	opts.Config.KubeConfig = configlib.GetConfigOrDie()
+
+	if opts.InstallCRDs {
+		if err := install.NewInstaller(opts.Config.KubeConfig).Install(&InstallStrategy{crds: inject.Injector.CRDs}); err != nil {
+			glog.Fatalf("Could not create CRDs: %v", err)
+		}
+	}
+
+	opts.Config.TargetNamespace = metav1.NamespaceAll
```
[the remark this answers is not in the document]
Comment (shashidharatd) [reply cmd/controller-manager/app/controller-manager.go]: Updated.

**Comment by shashidharatd** [comment]
> I think there's still value in being able to distribute kubefed2, since the UX for cli usage will be simpler. Is there still value in being able to distribute controller-manager separately, or could hyperfed replace it entirely?

Agree with you on distributing kubefed2. its a client tool and we might need to distribute this tool for Linux and Darwin (lets leave it as instruction to build it for now). I do feel there is no need to distribute controller-manager separately and can be taken over by hyperfed, like we do it in the federation-v2 docker image in this pr.

Also @marun, have addressed the comments. can you ptal, Thanks!

**Comment by marun** [comment]
Nice work!

/lgtm

**Pull request merged**